source/Company.py

```diff
@@ -1,53 +1,184 @@
+        else:
+            self.name = name
+        if check_positive(stocks_num) is False or check_int(stocks_num) is False:
+            raise ValueError("Error! Number of stocks must be positive.")
+        else:
+            self.stocks_num = stocks_num
+        if not check_positive(stock_price):
+            raise ValueError("Error! Price of stock must be positive.")
+        else:
+            self.stock_price = stock_price
+        if not check_string(comp_type):
+            raise ValueError("Error! Company type starts with capital letter.")
+        else:
+            self.comp_type = comp_type
 
     def net_worth(self):
-
-        pass
+        '''
+        Net worth of a company: as simple as that
+        '''
+        return self.stock_price*self.stocks_num
 
     def set_name(self, name):
-
-        pass
+        '''
+        Updates the company name, as long as its valid
+        '''
+        if name == self.name or check_string(name) is False:
+            return False
+        else:
+            self.name = name
+            return True
 
     def set_stocks_num(self, stocks_num):
-
-        pass
+        '''
+        Updates the number of stocks, as long as its valid
+        '''
+        if check_positive(stocks_num) is False or check_int(stocks_num) is False:
+            return False
+        else:
+            net_worth = self.net_worth()
+            self.stocks_num = stocks_num
+            self.stock_price = net_worth/self.stocks_num
+            return True
 
     def set_stock_price(self, stock_price):
-
-        pass
+        '''
+        Updates the price of a stock, as long as its valid
+        '''
+        if stock_price > self.net_worth() or check_positive(stock_price) is False:
+            return False
+        else:
+            net_worth = self.net_worth()
+            self.stock_price = stock_price
+            self.stocks_num = int(net_worth//self.stock_price)
+            return True
 
     def set_comp_type(self, comp_type):
-
-        pass
+        '''
+        Updates the type of the copmany, as long as its valid
+        '''
+        if not check_string(comp_type):
+            return False
+        else:
+            self.comp_type = comp_type
+            return True
 
     def update_net_worth(self, net_worth):
-
-        pass
+        '''
+        Updates the networth of the company
+        '''
+        if not check_positive(net_worth):
+            return False
+        else:
+            self.stock_price = net_worth / self.stocks_num
+            return True
 
     def add_stocks(self, number):
-
-        pass
+        '''
+        The function get int (numeber) and add it
+        to the current number of stocks
+        '''
+        if check_int(number) is False:
+            return False
+        new_stock_num = self.stocks_num + number
+        if new_stock_num <= 0:
+            return False
+        else:
+            self.stocks_num = new_stock_num
+            return True
```

Comment:
I see 2 False returns, this can be reduced to one.

Reply:
fixed